pkg/plugins/pg/README.md

```diff
@@ -1,144 +1,149 @@
-then will call `Ping` and test the connection. If the `Ping` fails, the `Open`
-method will fail and the Connector will not be ready to Read. 
+then will call `Ping` to test the connection. If the `Ping` fails, the `Open`
+method will fail and the Connector won't be started.
+
+Upon starting, the source takes a snapshot of a given table in the database, 
+then switches into CDC mode. In CDC mode, the plugin reads from a buffer of 
+CDC events.
+
+## Snapshot Capture
+When the connector first starts, snpashot mode is enabled. The connector
+acquires a read-only lock on the table, and then reads all rows of the table 
+into Conduit. Once all of the rows in that initial snapshot are read, then the 
+connector releases it's lock and switches into CDC mode. 
+
+This behavior is enabled by default, but can be turned off by adding 
+`"snapshot":"off"` to the Source configuration.
 
 ## Change Data Capture
-This connector implements CDC features for PostgreSQL by reading WAL log events 
-into a buffer that is checked on each Read request after the initial table Rows 
-have been read.
+This connector implements CDC features for PostgreSQL by reading WAL events 
+into a buffer that is checked on each call of `Read` after the initial snapshot
+has occurred. If there is a record in the buffer, it grabs and returns that 
+record. If it's empty, it returns `ErrEndData` signaling that Conduit should 
+backoff-retry.
 
-## CDC  Configuration
-When Open is called, the connector will attempt to start all necessary 
-connections and will run some initial setup commands to create the logical 
-replication slots it needs to run and start its own subscription. 
+ This behavior is enabled by default, but can be turned off by adding 
+ `"cdc": "off"` to the Source configuration.
 
-The connector user specified in the connection URL must have sufficient 
-privileges to run all of these commands or it will fail.
+### CDC  Configuration
+When the connector switches to CDC mode, it attempts to start all necessary 
+connections and runs the initial setup commands to create its logical 
+replication slots and publications. It will connect to an existing slot if one
+with the configured name exists.
 
-If the `cdc` field in the plugin.Configuration is set to any truthy value, it 
-will be enabled. 
+The Postgres user specified in the connection URL must have sufficient 
+privileges to run all of these setup commands or it will fail.
 
 Publication and slot name are user configurable, and must be correctly set. 
 The plugin will do what it can to be smart about publication and slot 
 management, but it can't handle everything.
 
-If a replication_url is provided, it will be used for CDC features instead of 
-the url. If no replication_url is provided, but cdc is enabled, then it will 
+If a `replication_url` is provided, it will be used for CDC features instead of 
+the url. If no `replication_url` is provided, but cdc is enabled, then it will 
```

Comment:
Not related to the docs, but to the reasoning behind this parameter - is there a use case when someone would want to provide 2 different URLs, one in `url` and one in `replication_url`?

Reply:
There was, but it will be refactored out with the plugin SDK transition.